src/Symfony/Component/Lock/Store/DoctrineDbalStore.php

```diff
@@ -92,8 +92,16 @@ public function save(Key $key)
                 ParameterType::STRING,
             ]);
         } catch (TableNotFoundException $e) {
-            $this->createTable();
-            $this->save($key);
+            if (!$this->conn->isTransactionActive() || $this->driverSupportsTableCreationInTransaction()) {
+                $this->createTable();
+            }
+            $this->conn->executeStatement($sql, [
```

Comment:
what if a `DBALException` is thrown here? the logic performed at line 107 is not performed anymore right?

Reply:
Yeah, I guess we need to double wrap in try/catch.

Reply:
Updated code.